test/e2e/apimachinery/execwithoptions_stress.go

```diff
@@ -0,0 +1,175 @@
+		}
+
+		ginkgo.By("waiting for busybox's availability")
+		err = e2epod.WaitForPodNameRunningInNamespace(ctx, f.ClientSet, pod.Name, f.Namespace.Name)
+		if err != nil {
+			e2edebug.DumpAllNamespaceInfo(ctx, f.ClientSet, f.Namespace.Name)
+			framework.Failf("unable to wait for busybox pod to be running and ready: %v", err)
+		}
+	})
+
+	ginkgo.It("works under load", func(ctx context.Context) {
+		start := time.Now()
+		duration := 3 * time.Minute
+		workers := 20
+
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		var wg sync.WaitGroup
+		wg.Add(workers)
+		for i := 0; i < workers; i++ {
+			go func(worker int) {
+				ginkgo.By(fmt.Sprintf("Worker #%d started.", worker))
+				defer wg.Done()
+				defer ginkgo.GinkgoRecover()
+				defer func() {
+					// Here we detect failures, do
+					// something, then pass that failure on
+					// to GinkgoRecover.
+					if r := recover(); r != nil {
+						// Notify other workers that they can stop prematurely.
+						cancel()
+						ginkgo.By(fmt.Sprintf("Worker #%d failed.", worker))
+						panic(r)
+					}
+					ginkgo.By(fmt.Sprintf("Worker #%d completed successfully.", worker))
+				}()
+
+				// Busy loop and check as often as possible
+				// during the entire test runtime until the
+				// time runs out, the test gets interrupted
+				// (parent context), or some other worker fails
+				// (our context).
+				for i := 0; time.Now().Sub(start) <= duration && ctx.Err() == nil; i++ {
+					data := generateRandomBytes(102400)
+					stdout, _, err := e2epod.ExecWithOptions(f, e2epod.ExecOptions{
+						Command:            []string{"cat", "-"},
+						Namespace:          f.Namespace.Name,
+						PodName:            pod.Name,
+						ContainerName:      "write-pod",
+						Stdin:              bytes.NewBuffer(data),
+						CaptureStdout:      true,
+						CaptureStderr:      true,
+						PreserveWhitespace: false,
+						Quiet:              true, // Avoid dumping this struct, which would include all of the stdin buffer..
+					})
+					if err != nil {
+						framework.Failf("attempt #%d in worker #%d: error of ExecWithOptions: %v", i, worker, err)
+					}
+					stdout_bytes := []byte(stdout)
+					if diff := cmp.Diff(data, stdout_bytes); diff == "" {
+						framework.Failf("attempt #%d in worker #%d: wrong stdout found:\nlen(data):\n%v\nlen(stdout):\n%v\n", i, worker, len(data), len(stdout_bytes))
+					}
+				}
+			}(i)
+		}
+
+		// Wait for all workers to succeed or fail.
+		wg.Wait()
+	})
+
+	ginkgo.PIt("can transfer any ASCII character", func(ctx context.Context) {
+		failedCharacters := transferEveryAsciiChar(f, pod.Name)
+		if failedCharacters != nil {
+			framework.Failf("there are failed transfers: %v", failedCharacters)
+		}
+
+	})
+})
+
+// transferEveryAsciiChar transfers all ascii characters one by one from 0-255 and returns ascii characters that were not sucessfully transfered
+func transferEveryAsciiChar(f *framework.Framework, podName string) []byte {
```

Comment:
This `transferEveryAsciiChar` is the core logic of the `can transfer any ASCII character` test. It would be better to have the for loop here directly on the test.

What can be moved into a local function is the `e2epod.ExecWithOptions` invocation. Then `works under load` can also call it without having to duplicate the invocation and error checking.

Reply:
fixed!